File: Webpage/server/webpage.server/webpage.server/src/main/java/abbruzzese/webpage/webpage/server/webpage/Repository/ProductRepository.java
package abbruzzese.webpage.webpage.server.webpage.Repository;

import java.util.List;
import java.util.Optional;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.CrudRepository;
import org.springframework.data.repository.query.Param;
import abbruzzese.webpage.webpage.server.webpage.entities.Product;

public interface ProductRepository extends CrudRepository<Product,String> {

 Optional<Product> findByNombre(String Nombre);

@Query ("SELECT p FROM Product p WHERE p.Categoria = :categoria")
List<Product> FindByCategoria(@Param("categoria") String categoria);

}

File: PHP_API/REPOSITORY/ProductRepository.php
<?php

namespace App\Repository;

use App\Entity\Product;
use Doctrine\ORM\EntityRepository;

class ProductRepository extends EntityRepository{

    public function findByNombre($nombre):?Product
    {
        return $this->findOneBy(['nombre' => $nombre]);
    }

    public function findByCategoria(string $categoria): array
    {
        $dql = "SELECT p FROM App\Entity\Product p WHERE p.categoria = :categoria";
        
        return $this->getEntityManager()
                    ->createQuery($dql)
                    ->setParameter('categoria', $categoria)
                    //->getOneOrNullResult();  // Obtiene un solo producto o null
                    ->getResult();
    }



}

?>